matdeeplearn/trainers/property_trainer.py

```diff
@@ -51,8 +51,10 @@ def train(self):
         # Start training over epochs loop
         # Calculate start_epoch from step instead of loading the epoch number
         # to prevent inconsistencies due to different batch size in checkpoint.
+    
         start_epoch = self.step // len(self.train_loader)
-        for epoch in range(start_epoch, self.max_epochs):
+        
+        for epoch in range(start_epoch, start_epoch + self.max_epochs):
```

Comment:
I see, maybe we can talk more in person about this? From what I understand, you're saying that essentially `max_epochs` represents the number of epochs that we want to run per task. I was under the impression that this should be for the total number of epochs over the course of the entire training, but maybe that doesn't really matter as much? If you look here, this is how OCP does it: https://github.com/Open-Catalyst-Project/ocp/blob/main/ocpmodels/trainers/energy_trainer.py#L175

It sounds like you just want a way to break up the task, right? So maybe we have an optional param called `max_checkpoint_epochs` or something like that which can be specified in the config, and if that's included then we only train for that amount. If it's not specified, then it continues to train normally.

So that way if a job stops unexpectedly, it could resume using the load_checkpoint feature without the user having to figure out how many more epochs to do in order for it to run for the total amount of time they want, if that makes sense.

Reply:
Sounds like a good idea, I'll implement this.